### PR TITLE
GC: Update Sweep config options to default enabled

### PR DESCRIPTION
**Pull request by markfields**
Fixes [AB#4660](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4660)

## Description

Sweep can be enabled via a GC Option passed by the app or container implementation, we don't need additional configs guarding the functionality.  Rather, we want these configs to remain only to disable Sweep for DataStores and/or AttachmentBlobs if needed.

### CODE REVIEW

File: packages/runtime/container-runtime/src/gc/gcDefinitions.ts [hunk 1]
```diff
@@ -43,29 +43,28 @@ export const gcTombstoneGenerationOptionName = "gcTombstoneGeneration";
...
 export const throwOnTombstoneUsageKey = "Fluid.GarbageCollection.ThrowOnTombstoneUsage";
-// Feature gate to enable GC version upgrade.
+/** Config key to enable GC version upgrade. */
 export const gcVersionUpgradeToV3Key = "Fluid.GarbageCollection.GCVersionUpgradeToV3";
-// Feature gate to enable GC sweep for datastores.
-// TODO: Remove Test from the flag when we are confident to turn on sweep
-export const sweepDatastoresKey = "Fluid.GarbageCollection.Test.SweepDataStores";
-// Feature gate to enable GC sweep for attachment blobs.
-export const sweepAttachmentBlobsKey = "Fluid.GarbageCollection.Test.SweepAttachmentBlobs";
+/** Config key to enable GC sweep for datastores. */
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/gc/gcDefinitions.ts:64]:
```suggestion
/** Config key to disable GC sweep for datastores. */
```

File: packages/runtime/container-runtime/src/gc/gcDefinitions.ts [hunk 2]
```diff
@@ -43,29 +43,28 @@ export const gcTombstoneGenerationOptionName = "gcTombstoneGeneration";
...
+export const noSweepDatastoresKey = "Fluid.GarbageCollection.NoSweepDataStores";
+/** Config key to enable GC sweep for attachment blobs. */
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/gc/gcDefinitions.ts:66]:
```suggestion
/** Config key to disable GC sweep for attachment blobs. */
```

File: packages/runtime/container-runtime/src/gc/gcDefinitions.ts [hunk 3]
```diff
@@ -43,29 +43,28 @@ export const gcTombstoneGenerationOptionName = "gcTombstoneGeneration";
...
-// Feature gate to enable GC version upgrade.
+/** Config key to enable GC version upgrade. */
 export const gcVersionUpgradeToV3Key = "Fluid.GarbageCollection.GCVersionUpgradeToV3";
-// Feature gate to enable GC sweep for datastores.
-// TODO: Remove Test from the flag when we are confident to turn on sweep
-export const sweepDatastoresKey = "Fluid.GarbageCollection.Test.SweepDataStores";
-// Feature gate to enable GC sweep for attachment blobs.
-export const sweepAttachmentBlobsKey = "Fluid.GarbageCollection.Test.SweepAttachmentBlobs";
+/** Config key to enable GC sweep for datastores. */
+export const noSweepDatastoresKey = "Fluid.GarbageCollection.NoSweepDataStores";
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/gc/gcDefinitions.ts:65]:
```suggestion
export const disableDatastoreSweepKey = "Fluid.GarbageCollection.DisableDataStoreSweep";
```

File: packages/runtime/container-runtime/src/gc/gcDefinitions.ts [hunk 4]
```diff
@@ -43,29 +43,28 @@ export const gcTombstoneGenerationOptionName = "gcTombstoneGeneration";
...
+/** Config key to enable GC sweep for attachment blobs. */
+export const noSweepAttachmentBlobsKey = "Fluid.GarbageCollection.NoSweepAttachmentBlobs";
```
[the remark this answers is not in the document]
Comment (agarwal-navin) [reply packages/runtime/container-runtime/src/gc/gcDefinitions.ts:67]: To keep it consistent with other disable configs + it sounds slightly better.

File: packages/runtime/container-runtime/src/gc/gcEarlyAdoption.md [hunk 5]
```diff
@@ -89,12 +89,14 @@ Note: The Summarizer client will _never_ throw on usage or load of a Tombstoned
...
 
--   GC Option `gcSweepGeneration` must be set, and the persisted value must match the current value in the code
--   Each of these two Config Settings must be set to `true` in the session:
-    -   `Fluid.GarbageCollection.Test.SweepDataStores`
-    -   `Fluid.GarbageCollection.Test.SweepAttachmentBlobs`
+```ts
+gcSweepGeneration: 0;
+```
+
+The exception would be if you ever bumped `gcTombstoneGeneration` - in that case, set `gcSweepGeneration` to `1` or higher.
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/gc/gcEarlyAdoption.md:98]: This is confusing. I would explain what the number 0 means here and when to set it to a higher value.

Then follow that up with a note that talks about its interaction with `gcTombstoneGeneration` and how there is a special case when `gcTombstoneGeneration` is > 0 (?).

**Review by agarwal-navin — APPROVED** [review]

File: packages/runtime/container-runtime/src/gc/gcEarlyAdoption.md (hunk 5 above)
Comment (markfields) [reply packages/runtime/container-runtime/src/gc/gcEarlyAdoption.md:98]: Yeah, in writing these docs I realized how random / non-intuitive this all is.  I'll try to reword it.

File: packages/test/test-utils/src/TestSummaryUtils.ts [hunk 6]
```diff
@@ -136,9 +140,18 @@ export async function createSummarizer(
...
  */
-export async function summarizeNow(summarizer: ISummarizer, reason: string = "end-to-end test") {
-	const result = summarizer.summarizeOnDemand({ reason });
+export async function summarizeNow(
+	summarizer: ISummarizer,
+	inputs: string | IOnDemandSummarizeOptions = "end-to-end test",
+) {
+	const options: IOnDemandSummarizeOptions =
+		typeof inputs === "string" ? { reason: inputs } : inputs;
+	const result = summarizer.summarizeOnDemand(options);
```
Comment (markfields) [inline packages/test/test-utils/src/TestSummaryUtils.ts:154]: @agarwal-navin is this ok long-term or should I do a follow-up that replaces this with just an `options` param?

**Pull request merged**